worker/upload.go

```diff
@@ -1055,7 +1055,14 @@ func (s *slabUpload) receive(resp sectorUploadResp) (bool, bool) {
 	}
 
 	// redundant sectors can't complete the upload
-	if sector.uploaded.Root != (types.Hash256{}) {
+	if sector.isUploaded() {
+		// release the candidate
+		for _, candidate := range s.candidates {
+			if candidate.req == req {
```

Comment:
We could back link the candidate on the request. There's no harm in doing so imo, but doesn't really matter much. I see how this PR improves things but at the same time I think it's a bit of a roundabout way of achieving what we want, should we consider resorting the candidates in certain spots?

Reply:
I'd rather avoid the potential issues arising from back referencing tbh. Before you know it you forget to update the reference and end up in a world of pain due to the inconsistency. 

We already take queue length into account in the estimate but if we don't wait for the candidate to release itself, there might be a race with the tracking of the elapsed time because the estimate might not be updated before we reschedule the candidate.
Overall it's just a nicer architecture to have each candidate manage itself in `receive` rather than having other candidates interfere.